GithubRepo/Data/DataSource/GithubDataSource.swift

```diff
@@ -0,0 +1,12 @@
+//
+//  GithubDataSource.swift
+//  GithubRepo
+//
+//  Created by Marcos Alves on 15/10/21.
+//
+
+import Foundation
+
+protocol GithubDataSource {
+    func getRepositories(with query: String, completion: @escaping (Result<[RepositoryResponse], Error>) -> Void)
```

Comment:
Your dataSource can always work with business objects, so the repository will be independent of responses model
```suggestion
    func getRepositories(with query: String, completion: @escaping (Result<[Repository], Error>) -> Void)
```